generator/config.json

```diff
@@ -629,6 +725,9 @@
         },
         "azuread": {
           "$ref": "#/$defs/AzureAdconfig"
+        },
+        "google_iam": {
+          "$ref": "#/$defs/Config"
```

Comment:
This is the problematic bit, the ref is resolves to the top-level config, causing endless recursion.

The generator for this schema doesn't handle duplicate naming of structs properly, causing this problem. It refers to this simple struct: https://github.com/prometheus/prometheus/blob/main/storage/remote/googleiam/googleiam.go#L28

We can easily account for this in the generator.